### PR TITLE
Elements: Add UI for button elements

### PR DESCRIPTION
**Pull request by scruffian**
## What?
#41246 introduced support for the `button` element in `theme.json`. This adds a UI to make it possible to edit button elements in Global Styles. 

## Why?
To allow users to edit their button styles.

## How?
Duplicating existing global styles controls.

## Testing Instructions
1. In the Site Editor, open Global Styles
2. Open the Typography panel
3. You should see an option to modify the typography options for buttons
4. Open the Colors panel
5. You should see an option to modify the color options for buttons
6. Modifications in both panels should update the preview and when saved the frontend.

## Screenshots or screencast 
Typography

https://user-images.githubusercontent.com/107534/174039260-65f705c3-29fd-41b1-961e-924731808054.mp4

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/ui.js
Comment (adamziel) [inline packages/edit-site/src/components/global-styles/ui.js]: How about extracting these declarations to a new file called `elements.js` that would export a single variable, such as below? This way it you could manage it from a single file.

```js
export const GlobalStylesElements = [
    {
        name: 'button',
        editTypography: {
		title: __( 'Buttons' ),
		description: __( 'Manage the fonts and typography used on buttons.' ),
        },
        editColors: {
		title: __( 'Buttons' ),
		description: __( 'Set the default colors used for buttons across the site.' ),
        	component: ScreenButtonColor,
        }
    }
]
```

File: packages/blocks/src/api/constants.js
Comment (draganescu) [inline packages/blocks/src/api/constants.js]: I found this confusing. The best way wold be to follow CSS and use `[ 'elements', 'button', 'background', 'color' ]` but that is outside the scope.

Comment (scruffian) [reply packages/blocks/src/api/constants.js]: This is the way that the data is structured in Global styles - properties are grouped by `color`, `typography`, `spacing` etc. It doesn't map onto CSS.

File: packages/edit-site/src/components/global-styles/style.scss
Comment (getdave) [inline packages/edit-site/src/components/global-styles/style.scss]: What is this 8px about? Why should I note modify this value? How is it related to the layout and/or other values?

File: packages/edit-site/src/components/global-styles/style.scss
Comment (getdave) [inline packages/edit-site/src/components/global-styles/style.scss]: How come we are using `rgb` notation here?

File: packages/edit-site/src/components/global-styles/ui.js
Comment (getdave) [reply packages/edit-site/src/components/global-styles/ui.js]: I think we should create a _follow up_ Issue to do just this. @scruffian or @draganescu may be the best persons to do this as he has his head in the right code. We shouldn't delay on this as it's inevitable we will want to extend with more controls in future. I propose making the follow up Issue now with clear instructions on the likely path forward and references to the appropriate files.

**Review by getdave — COMMENTED** [review]
Nice work on getting this PR up 👏 

I gave it a spin with TT2 and I'd say it's 80% of the way. I noticed:

- font size has no effect
- colors have no effect

https://user-images.githubusercontent.com/444434/174748862-6503c6e2-af88-4478-b0f7-0cc1cafdafe7.mov

File: packages/edit-site/src/components/global-styles/style.scss
Comment (draganescu) [reply packages/edit-site/src/components/global-styles/style.scss]: I think it is copy pasta, I can check out what variable serves me.

File: packages/edit-site/src/components/global-styles/ui.js
Comment (draganescu) [reply packages/edit-site/src/components/global-styles/ui.js]: Same as @getdave suggested once we merge this duplication I think it'll become even easier to concoct an abstraction to simplify.

File: packages/edit-site/src/components/global-styles/style.scss
Comment (draganescu) [reply packages/edit-site/src/components/global-styles/style.scss]: 🤷🏻 I can't even remember where did I come up with that from. But it looks nice 🏃🏻 
Will fix.

**Review by getdave — COMMENTED** [review]
Reviewed this again but this time with Empty Theme (incidentally I notcied the reason I used TT2 on my first review was because that was shown in the demo screenrecording on the PR).

Unfortunately I still get the same results.

https://user-images.githubusercontent.com/444434/174995081-f10241fb-b32d-4781-8c64-10fca87019d3.mov

I believe the issue is that CSS styles are overloading the global style rules:

Screen Shot 2022-06-22 at 10 33 05

**Comment by getdave** [comment]
I can't seem to dismiss my previous reviews. 

However, having spoken with @draganescu he's explained that https://github.com/WordPress/gutenberg/pull/41822 actually fixes a number of the CSS specificity Issues I identified with this PR.

Therefore we should wait for that PR to be merged and this PR to be rebased before attempting further reviews.

**Comment by scruffian** [comment]
Thanks for picking this up and running with it @draganescu. LGTM

**Review by getdave — APPROVED** [review]
I tested this again and it now seems to work a lot better. I can adjust both color and typography for buttons and it propagates.

https://user-images.githubusercontent.com/444434/175060580-b5b30c11-d136-4b96-9216-ebe254b61a26.mov



This only works with Empty Theme for now because other Themes are still adding styles for button elements in CSS rather than in the new improved `.json` way.

Nice work :shipit:

File: packages/edit-site/src/components/global-styles/ui.js
Comment (getdave) [reply packages/edit-site/src/components/global-styles/ui.js]: I think @adamziel took this up in https://github.com/WordPress/gutenberg/pull/41841

**Pull request merged**
